common/ntp/ntp.go

```diff
@@ -0,0 +1,46 @@
+package ntp
+
+import (
+	"time"
+
+	beevik_ntp "github.com/beevik/ntp"
+	"github.com/pkg/errors"
+)
+
+const (
+	// tolerance range of local clock with NTP time server
+	toleranceRangeWarning = time.Duration(10 * time.Second)
+	toleranceRangeError   = time.Duration(30 * time.Second)
+)
+
+var (
+	errDriftTooMuch = errors.New("local time drift off ntp server more than 30 seconds")
+	errDriftInRange = errors.New("local time drift off ntp server more than 10 seconds")
+)
+
+// CheckLocalTimeAccurate returns whether the local clock accurate or not
+func CheckLocalTimeAccurate(ntpServer string) (bool, error) {
+	response, err := beevik_ntp.Query(ntpServer)
+	// failed to query ntp time
+	if err != nil {
+		return false, err
+	}
+	// drift too much
+	if response.ClockOffset > toleranceRangeError {
+		return false, errDriftTooMuch
```

Comment:
the return value for this func is a bit messy. error should only be used to indicate something wrong happened in the normal operations of the func, causing failure to do the work and return a value. The errDriftTooMuch and errDriftInRange are actually just different return values, not errors. It would be better to have a enum for the return value: <Normal, DriftTooMuch, DriftInRange>, rather than piggy back on error for value return.

Reply:
I felt that's a bit over-engineering for such a simple func. we used error to provide additional info on why the function failed.